### PR TITLE
Fixes MDAnalysis.analysis.density.convert_density() method has an invalid default unit value

### LINKED ISSUE #4829: MDAnalysis.analysis.density.convert_density() method has an invalid default unit value
## Expected behavior ##

This is a small one.  Some of the methods in `MDAnalysis.analysis.density` have invalid defaults.

`density.convert_density()` has a default of `unit='Angstrom'`, but this is not a permitted value.  As a result, if  `density.convert_density()` is run with no options specified, it fails with an error.  From the documentation, I suspect the permitted default would be `unit='Angstrom^{-3}'`

## Code to reproduce the behavior ##

```
from MDAnalysis.analysis.density import DensityAnalysis
u = Universe(TPR, XTC) # some universe with a thing you want to measure, in my case I was looking at sodium ions but I've used water for the example.
ow = u.select_atoms("name OW")
D = DensityAnalysis(ow, delta=1.0)
D.run()
D.density.convert_density()
```

expected outcome:  it runs a density calculation

actual outcome:  

```
ValueError: The name of the unit (Angstrom supplied) must be one of:
dict_keys(['Angstrom^{-3}', 'A^{-3}', 'Å^{-3}', 'nm^{-3}', 'nanometer^{-3}', 'Molar', 'SPC', 'TIP3P', 'TIP4P', 'water'])
```

## Current version of MDAnalysis ##

MDAnalysis 2.7.0
Python 3.10.14
Ubuntu 20.04.4

### PR DESCRIPTION
**Pull request by tanishy7777**
Fixes #4829 

Changes made in this Pull Request:
 - Changed the default unit in the convert_density() function to `Angstrom^{-3}` from `Angstrom`
 - Added new tests to check if the changes to the code(default unit) solve the issue in the `testsuite/MDAnalysisTests/analysis/test_density.py`

Before 
```python
def convert_density(self, unit='Angstrom'):
```

After
```python
def convert_density(self, unit='Angstrom^{-3}'):
```

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4833.org.readthedocs.build/en/4833/

### CODE REVIEW

**Comment by tanishy7777** [comment]
## Before the changes the test was failing as shown below
image
image
image

## After making the changes
image

**Review by RMeli — APPROVED** [review]
Thanks @tanishy7777, LGTM!

**Comment by RMeli** [comment]
Conflicts needs to be resolved here too, otherwise it's good to go.

File: testsuite/MDAnalysisTests/analysis/test_density.py [hunk 1]
```diff
@@ -384,6 +384,12 @@ def test_warn_results_deprecated(self, universe):
         with pytest.warns(DeprecationWarning, match=wmsg):
             assert_equal(D.density.grid, D.results.density.grid)
 
+    def test_density_analysis_invalid_unit(self):
```
Comment (RMeli) [inline testsuite/MDAnalysisTests/analysis/test_density.py]:
```suggestion
    def test_density_analysis_conversion_default_unit(self):
```
? Sorry for not noticing this before.

**Review by RMeli — CHANGES_REQUESTED** [review]

File: testsuite/MDAnalysisTests/analysis/test_density.py (hunk 1 above)
Comment (tanishy7777) [reply testsuite/MDAnalysisTests/analysis/test_density.py]: > ? Sorry for not noticing this before.

np. I have added the changes

**Review by RMeli — APPROVED** [review]

**Pull request merged**